lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -269,7 +269,7 @@ def initconfig_hardware_entries(self):
                 cmake_cache_path("HIP_CXX_COMPILER", "{0}".format(self.spec["hip"].hipcc))
             )
             archs = self.spec.variants["amdgpu_target"].value
-            if archs != "none":
+            if archs[0] != "none":
```

Comment:
Note that the `ROCmPackage` declares a conflict for `amdgpu_target=none` when `+rocm`, so in fact this `if` statement is entirely unnecessary (at least when using the provided ROCm constructs as-is). CUDA is not handled in the same way though, so it makes sense why you'd have to be careful there. Anyhow this is no worse, and as it is `amdgpu_target` is multi-valued like `cuda_arch`, so if the conflict were relaxed somewhere, this would still be a good check.